src/Sarif/CodeGenHints.json

```diff
@@ -1,156 +1,166 @@
 {
   "Rule.Options": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.DictionaryHint, Microsoft.Json.Schema.ToDotNet"
+      "kind": "DictionaryHint"
     }
   ],
   "AnnotatedCodeLocation.Properties": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.DictionaryHint, Microsoft.Json.Schema.ToDotNet"
+      "kind": "DictionaryHint" 
     }
   ],
   "file": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.ClassNameHint, Microsoft.Json.Schema.ToDotNet",
-      "className": "FileData"
+      "kind": "ClassNameHint",
+      "arguments": {
+        "className": "FileData"
```

Comment:
The rest of the changes involve wrapping the other properties of the hint object -- which Json.NET would previously have directly deserialized into the specified type -- in a property bag named `arguments`. JSchema constructs each type "by hand" from the contents of the property bag.